### PR TITLE
Fix determination of copyable layout for compressed textures in Vulkan

### LINKED ISSUE #86628: Saving a VRAM Compressed image is broken in 4.3-dev after RenderingDevice refactor
### Tested versions

The issue occurs in the latest master (13a0d6e9b253654f5cc2a44f3d0b3cae10440443), it occurs with #83452's merge commit (3a8524dd923e9d9a79e4979a47fd427388ea7010), and all commits between that I tested. The issue does not occur with commits before #83452's merge commit, including 4.2-stable.

### System information

macOS 14.2.1, Apple M1 Pro

### Issue description

Saving VRAM Compressed images is broken in 4.3-dev as of #83452. I noticed this bug when I tried to export a GLTF scene from Godot and the image data was extremely distorted.

### Steps to reproduce

1. Run the minimal reproduction project, which has icon.svg imported as VRAM Compressed, and has this code:

```gdscript
const ICON = preload("res://icon.svg")

func _ready():
	ICON.get_image().save_png("res://output.png")
```

2. With the latest master, 83452, and commits between, this image is output:

output

3. With 4.2-stable and commits before 83452, this image is output:

output

### Minimal reproduction project (MRP)

[imgsave.zip](https://github.com/godotengine/godot/files/13797504/imgsave.zip)

### PR DESCRIPTION
**Pull request by RandomShaper**
Fixes #86628.

### CODE REVIEW

File: drivers/vulkan/rendering_device_driver_vulkan.cpp [hunk 1]
```diff
@@ -805,8 +805,11 @@ void RenderingDeviceDriverVulkan::texture_get_copyable_layout(TextureID p_textur
 			h = MAX(1u, h >> 1);
 			d = MAX(1u, d >> 1);
 		}
-		r_layout->size = get_image_format_required_size(tex_info->rd_format, w, h, d, 1);
-		r_layout->row_pitch = r_layout->size / (h * d);
+		uint32_t bw = 0, bh = 0;
+		get_compressed_image_format_block_dimensions(tex_info->rd_format, bw, bh);
```
Comment (clayjohn) [inline drivers/vulkan/rendering_device_driver_vulkan.cpp:809]: It looks like this doesn't get used

Comment (RandomShaper) [reply drivers/vulkan/rendering_device_driver_vulkan.cpp:809]: `bh` is used in line 812.

**Review by clayjohn — APPROVED** [review]
Looks fine to me

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**